proposals/generic-attributes.md

```diff
@@ -0,0 +1,61 @@
+# Generic Attributes
+
+## Summary
+[summary]: #summary
+
+When generics were introduced in C# 2.0, attribute classes were not allowed to participate. We can make the language more composable by removing (rather, loosening) this restriction. The .NET Core runtime has added support for generic attributes. Now, all that's missing is support for generic attributes in the compiler.
+
+## Motivation
+[motivation]: #motivation
+
+Currently attribute authors can take a `System.Type` as a parameter and have users pass a `typeof` expression to provide the attribute with types that it needs. However, outside of analyzers, there's no way for an attribute author to constrain what types are allowed to be passed to an attribute via `typeof`. If attributes could be generic, then attribute authors could use the existing system of type parameter constraints to express the requirements for the types they take as input.
+
+## Detailed design
+[design]: #detailed-design
+
+The following section is amended: https://github.com/dotnet/csharplang/blob/main/spec/classes.md#base-classes
+
+> The direct base class of a class type must not be any of the following types: System.Array, System.Delegate, System.MulticastDelegate, System.Enum, or System.ValueType. ~~Furthermore, a generic class declaration cannot use System.Attribute as a direct or indirect base class.~~
+
+One important note is that the following section of the spec is *unaffected* when referencing the point of usage of an attribute, i.e. within an attribute list: https://github.com/dotnet/csharplang/blob/main/spec/types.md#type-parameters
+
+> A type parameter cannot be used anywhere within an attribute.
```

Comment:
Something about this spec line had me confused for a bit (as well as once before ~a year ago?), so I'm wondering if it needs to be rephrased to emphasize it refers to the attribute placement and not its definition.

Reply:
It does seem like it would be a little more clear to adjust the terminology to make attribute usages more distinct from their definitions.